### PR TITLE
Soft deprecate Sink

### PR DESCRIPTION
**Pull request by SUPERCILEX**
https://rust-lang.zulipchat.com/#narrow/stream/219381-t-libs/topic/Deprecate.20sink.3F

### CODE REVIEW

File: library/std/src/io/mod.rs [hunk 1]
```diff
@@ -273,13 +273,14 @@ pub use self::stdio::IsTerminal;
...
 pub use self::stdio::{_eprint, _print};
 #[stable(feature = "rust1", since = "1.0.0")]
+#[allow(deprecated_in_future)]
 pub use self::{
     buffered::{BufReader, BufWriter, IntoInnerError, LineWriter},
     copy::copy,
     cursor::Cursor,
     error::{Error, ErrorKind, Result},
     stdio::{stderr, stdin, stdout, Stderr, StderrLock, Stdin, StdinLock, Stdout, StdoutLock},
-    util::{empty, repeat, sink, Empty, Repeat, Sink},
```
Comment (Nilstrieb) [inline library/std/src/io/mod.rs:282]: why that?

Comment (SUPERCILEX) [reply library/std/src/io/mod.rs:282]: Mistake

**Comment by Mark-Simulacrum** [comment]
Nominating for libs-api to decide if we want to do this.

**Comment by Amanieu** [comment]
We discussed this in the libs-api meeting yesterday. We don't see any significant benefits to deprecating `Sink`. Instead it would be sufficient to just encourage the use of `Sink` when reading from the file is not required, and `Empty` if it is. The documentation for both should be updated to mention the other and under which circumstances it would be appropriate to use it.

**Pull request closed without merging**